### ISSUE TITLE
det_results/dance data

### ISSUE THREAD

**Issue by saijo0404** [Paulkie99/imm-jhse#2]
How to get the data in det_results/dance/ ?

**Comment by Paulkie99** [comment]
Hi @saijo0404, we used the detector from [ByteTrack](https://github.com/ifzhang/ByteTrack), which is also used by OC-SORT and UCMCTrack. For pretrained weights, see [the OC-SORT repo](https://github.com/noahcao/OC_SORT/blob/master/docs/MODEL_ZOO.md).

**Comment by saijo0404** [comment]
So you're using the OC-SORT repo's pre-trained weights to detect the images and get this data?

**Comment by Paulkie99** [comment]
Hi @saijo0404. Yes, detections are obtained using OC-SORT's pretrained weights, but we use IMM-JHSE to perform tracking and association of these detection ``measurements'', and not OC-SORT's method.

So, the `det_results` folder contains only raw detections. The `result_files` folder contains the results of tracking and association with IMM-JHSE.

**Comment by saijo0404** [comment]
Hi @Paulkie99  thanks for your reply.
Another question, is the gmc data obtained using get_gmc.py?

**Comment by Paulkie99** [comment]
Hey @saijo0404, the GMC files for the MOT17 and MOT20 datasets are provided by [UCMCTrack](https://github.com/corfyi/UCMCTrack). But for the DanceTrack and KITTI datasets, get_gmc.py is used.

**Comment by Paulkie99** [comment]
To clarify how detections are obtained:

### DanceTrack
Per the [DanceTrack README](https://github.com/noahcao/OC_SORT/blob/master/docs/GET_STARTED.md), the following command is used to evaluate OC-SORT:
`python tools/run_ocsort_dance.py -f exps/example/mot/yolox_dancetrack_test.py -c pretrained/bytetrack_dance_model.pth.tar -b 1 -d 1 --fp16 --fuse --test --expn $exp_name`

This executes the [run_ocsort_dance.py](https://github.com/noahcao/OC_SORT/blob/master/tools/run_ocsort_dance.py) script, which uses the [MotEvaluator](https://github.com/noahcao/OC_SORT/blob/master/tools/run_ocsort_dance.py#L73) class. To extract detections, the following alterations are made to the [MotEvaluator ](https://github.com/noahcao/OC_SORT/blob/master/yolox/evaluators/mot_evaluator_dance.py) class:

Create a folder and file to record detection results:
![image](https://github.com/user-attachments/assets/23a6e565-1469-431e-b714-3f892b26cb62)

Write detection results **before the OC-SORT tracker is updated**:
![image](https://github.com/user-attachments/assets/472a7d8c-067b-4ecd-9746-42755d8e6a33)

### KITTI 
We run [PermaTrack](https://github.com/TRI-ML/permatrack/blob/main/readme/GETTING_STARTED.md), **the detection results for the test set are identical to those provided by** [OC-SORT](https://github.com/noahcao/OC_SORT/tree/master/exps/permatrack_kitti_test).

### MOT Datasets
These are taken from [UCMCTrack](https://github.com/corfyi/UCMCTrack/tree/master/det_results).
